inductiva/resources/machines_base.py

```diff
@@ -285,7 +285,7 @@ def from_api_response(cls, resp: dict):
 
         return machine_group
 
-    def _can_start_resource(self) -> bool:
```

Comment:
why change the method to public?

Reply:
This function is only invoked during the start process and is not intended for user access. Therefore, it makes sense to keep it private. It was previously set as private and successfully passed the linter tests. I recommend reverting it to private.

Reply:
I don't see why this function shouldn't be accessible to users. Also, the method was not being tested previously, at all -- it was another one imported from the _benchmarks module. I can try again, but I believe reverting it to private won't pass the linter tests.

Reply:
> I don't see why this function shouldn't be accessible to users. Also, the method was not being tested previously, at all -- it was another one imported from the _benchmarks module. I can try again, but I believe reverting it to private won't pass the linter tests.

This behaviour will be controlled by a flag `wait_on_pending_quota` in the start method. The user just need to do mg.start(wait_on_pending_quota=True) instead of doing 
```
if mg.can_start_resource:
    mg.start()
```

As for testing. That will be hard. We need to mock the object creation to prevent the machinegroup being created in the back end. We then need to mock the get_quotas. After that we can test the method.

Reply:
I can add the tests if you want.

Reply:
Please check the final version of this PR, which includes tests using those mock objects.

Reply:
> > I don't see why this function shouldn't be accessible to users. Also, the method was not being tested previously, at all -- it was another one imported from the _benchmarks module. I can try again, but I believe reverting it to private won't pass the linter tests.
> 
> This behaviour will be controlled by a flag `wait_on_pending_quota` in the start method. The user just need to do mg.start(wait_on_pending_quota=True) instead of doing
> 
> ```
> if mg.can_start_resource:
>     mg.start()
> ```
> 
> As for testing. That will be hard. We need to mock the object creation to prevent the machinegroup being created in the back end. We then need to mock the get_quotas. After that we can test the method.

Please check the final version of this PR, which includes tests using those mock objects.

Reply:
> I can add the tests if you want.

It has already been done, thanks.